lib/util.js

```diff
@@ -128,6 +128,30 @@ exports.chooseFormat = (formats, options) => {
       format = formats[formats.length - 1];
       break;
 
+    case 'highestaudio':
+      formats = exports.filterFormats(formats, 'audio');
+      format = formats[0];
+      for (let f of formats) {
+        if (f.audioBitrate > highest.audioBitrate
+          || (f.audioBitrate === highest.audioBitrate && !f.encoding))
```

Comment:
In the case where two formats have the same audio quality, do we prefer the format that does not contain a video? If not, then I don't think we need this line

Same for the other case below with audio/video switched

Reply:
the problem i'm having is that most ppl using `highestaudio` and `highestvideo` will most likely merge them or u'd pick them 'cause ur not interested in the other and most likely removing it from the stream anyway
the other problem i'm personaly having is that using both `highestaudio` and a filter `audioonly` results in an error getting throw since some videos actually only support one format(for my e.g. itag 18)

Reply:
>the problem i'm having is that most ppl using highestaudio and highestvideo will most likely merge them or u'd pick them 'cause ur not interested in the other and most likely removing it from the stream anyway

Hmm, in that case I'd change the above description in the readme to

    `highestaudio`/`highestvideo` both prefer audio/video only respectively.


>the other problem i'm personaly having is that using both highestaudio and a filter audioonly results in an error getting throw since some videos actually only support one format(for my e.g. itag 18)

What's the error getting thrown? We should probably fix that, doesn't need to be in this PR though

Reply:
>> the other problem i'm personaly having is that using both highestaudio and a filter audioonly results in an error getting throw since some videos actually only support one format(for my e.g. itag 18)
>
>What's the error getting thrown? We should probably fix that, doesn't need to be in this PR though

did a typo there, was actually using `{filter: 'audioonly'}` which ended with `new Error('No formats found with custom filter');` ([L116](https://github.com/fent/node-ytdl-core/blob/master/lib/util.js#L116))
so no need to fix that...